testing/dart/canvas_test.dart

```diff
@@ -1031,6 +1031,38 @@ void main() async {
     canvas.restoreToCount(canvas.getSaveCount() + 1);
     expect(canvas.getSaveCount(), equals(6));
   });
+
+  test('TextDecoration renders non-solid lines', () async {
```

Comment:
This doesn't seem that bad, you made it seem like this would be pretty hard

Reply:
Easy to write, hard to run :)